<launch>
    <!--
    catch2_launch_file
    A required tag which will add a "result_file" argument for the launch file.
    This file will hold the results of the test.
    Description attribute of this tag is optional for XML launch files.
    -->
    <catch2_launch_file description='Integration test for each of the Teleop Nodes'/>

    <!-- Specific launch arguments can also be included at the user's discretion -->
    <arg 
        name='test_duration'
        default='5.0'
        description='Max length of test in seconds.'
    />

    <!-- Run the Teleop Nodes to be tested -->
    <!-- The point_stamped_mirror node -->
    <node name="point_stamped_mirror_node" pkg="unified_teleop" exec="point_stamped_mirror">
        <param from="$(find-pkg-share unified_teleop)/test/output_scheme_test.yaml" />
        <param name="input_device_config" value="input_mapping_test" />
        <remap from="desired_position" to="point_stamped_mirror/desired_position" />
    </node>

    <!--
    catch2_integration_test_node
    a wrapper around node which passes the "result_file" argument to Catch2.
    There should only be one integration test node. This node will shutdown
    the entire launch file when it exits.
    Specific parameters and other arguments can also be passed, like the
    "test_duration" example below.
    -->
    <catch2_integration_test_node
        pkg='unified_teleop'
        exec='integration_test_node'>
        <param name='test_duration' value='$(var test_duration)'/>
    </catch2_integration_test_node>
</launch>